nemo_skills/pipeline/eval.py

```diff
@@ -409,46 +410,160 @@ def eval(
+                        unmounted_done = pipeline_utils.get_unmounted_path(cluster_config, done_file)
+
+                        # Check if files exist
+                        if cluster_config["executor"] == "local":
+                            if not (Path(unmounted_output).exists() and Path(unmounted_done).exists()):
+                                all_done = False
+                                break
+                        else:
+                            if not (
+                                pipeline_utils.cluster_path_exists(cluster_config, unmounted_output)
+                                and pipeline_utils.cluster_path_exists(cluster_config, unmounted_done)
+                            ):
+                                all_done = False
+                                break
+
+                    if all_done:
+                        LOG.info(f"Skipping NVEmbed judge for {benchmark} - all output files and .done markers exist")
+                        judge_tasks = None
+                        continue
+
+                # Install required packages for NVEmbed evaluation
+                install_cmd = (
+                    "pip install -q -e /nemo_run/code && pip install -q datasets einops transformers==4.42.4 && "
+                    "python -c 'import torch; print(f\"PyTorch {torch.__version__} with CUDA {torch.version.cuda}, CUDA available: {torch.cuda.is_available()}\")'"
+                )
+
+                # Build list of (input_file, output_file) pairs to process
+                # Generation pipeline automatically merges chunks, so we just copy and evaluate merged files
+                files_to_process = []
+                if input_file is None:
+                    # Multiple seeds case: output-rs0.jsonl, output-rs1.jsonl, etc.
+                    input_dir = judge_pipeline_args.get("input_dir")
+                    num_seeds = judge_pipeline_args.get("num_random_seeds", 1)
+                    for seed in range(num_seeds):
+                        files_to_process.append(
+                            (f"{input_dir}/output-rs{seed}.jsonl", f"{output_dir_path}/output-rs{seed}.jsonl")
+                        )
+                else:
+                    # Single file case: output.jsonl
+                    files_to_process.append((input_file, f"{output_dir_path}/output.jsonl"))
+
+                # Create evaluation commands for all files
+                # Check generation's .done file first to ensure all chunks were merged successfully
+                eval_cmds = []
+                for src_file, dst_file in files_to_process:
+                    eval_cmd = (
+                        # First verify generation completed by checking for .done file
+                        f'if [ ! -f "{src_file}.done" ]; then '
+                        f'  echo "Error: Generation not complete: {src_file}.done not found"; '
```

Comment:
I don't think we need an explicit check, can just fail with normal error of not finding an input file in case previous job isn't finished. So I'd probably just run your command assuming it's there and if not it should be clear from error message that previous step has an issue